src/drivers/rc_input/RCInput.cpp

```diff
@@ -308,16 +308,17 @@ void RCInput::Run()
 		adc_report_s adc;
 
 		if (_adc_sub.update(&adc)) {
-			const unsigned adc_chans = sizeof(adc.channel_id) / sizeof(adc.channel_id[0]);
-
-			for (unsigned i = 0; i < adc_chans; i++) {
+			for (int i = 0; i < PX4_MAX_ADC_CHANNELS; ++i) {
 				if (adc.channel_id[i] == ADC_RC_RSSI_CHANNEL) {
+					float adc_volt = adc.raw_data[ADC_RC_RSSI_CHANNEL] *
+							 adc.v_ref[ADC_RC_RSSI_CHANNEL] /
+							 adc.resolution;
 
 					if (_analog_rc_rssi_volt < 0.0f) {
-						_analog_rc_rssi_volt = adc.channel_value[i];
+						_analog_rc_rssi_volt = adc_volt;
```

Comment:
This looks like it repeated the last sample.  Now it does not

Reply:
I cannot find those lines of code. Instead they already are.
https://github.com/SalimTerryLi/Firmware/blob/86808ed736efb40f1f832869de786ffedce3ed27/src/drivers/rc_input/RCInput.cpp#L317-L319
Not sure what is happening here.

Reply:
Here is the old code https://github.com/PX4/Firmware/blob/bcfc0e2f5625cc975ecbd304e9cfeaac51d362bf/src/drivers/rc_input/RCInput.cpp#L310-L326

It looks like it has been changed, and now does not act the same. @dagar ?

Reply:
I'm not sure what the question/confusion is. Here's the corresponding chunk of logic. https://github.com/PX4/Firmware/blob/4c1adc088fa8e7cc682e16582300cbf91cd84f67/src/drivers/rc_input/RCInput.cpp#L200-L214

It might be easier and simpler to do the adc_report copy and minimal work directly in `RCInput::fill_rc_in` when rssi hasn't been provided.

Reply:
> I'm not sure what the question/confusion is. Here's the corresponding chunk of logic.
> 
> https://github.com/PX4/Firmware/blob/4c1adc088fa8e7cc682e16582300cbf91cd84f67/src/drivers/rc_input/RCInput.cpp#L200-L214
> 
> It might be easier and simpler to do the adc_report copy and minimal work directly in `RCInput::fill_rc_in` when rssi hasn't been provided.

hard-coded base voltage which is used to calculate rssi. This may work, may not either.
But in this PR I only do the replacement of data source so this might not something to be discussed here?

Reply:
@davids5 Please forgive me, I'm really confused with what you said. In the `RCInput.cpp`, I only replace old ioctl call with new uORB message, no functional logic changed. In other word, this PR is not aimed to improve anything about RSSI, but only keep its existing function works with new uORB subscription. No doubt `adc.channel_value[i]` is equal to `adc_volt`.